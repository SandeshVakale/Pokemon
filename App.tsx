import React from 'react';
import {StatusBar, StyleSheet} from 'react-native';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {
  useFonts,
  Lexend_100Thin,
  Lexend_300Light,
  Lexend_400Regular,
  Lexend_500Medium,
  Lexend_600SemiBold,
  Lexend_700Bold,
  Lexend_800ExtraBold,
} from '@expo-google-fonts/lexend';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/lib/integration/react';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import Navigation from './src/navigation';
import {store, persistor} from './src/store';
import Loader from './src/components/loader';
export default function App() {
  let [fontsLoaded] = useFonts({
    Lexend_100Thin,
    Lexend_300Light,
    Lexend_400Regular,
    Lexend_500Medium,
    Lexend_600SemiBold,
    Lexend_700Bold,
    Lexend_800ExtraBold,
  });

  if (!fontsLoaded) {
    return <Loader />;
  }
  return (
    <GestureHandlerRootView style={styles.container}>
      <SafeAreaProvider>
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <StatusBar barStyle={'light-content'} />
            <Navigation />
          </PersistGate>
        </Provider>
      </SafeAreaProvider>
    </GestureHandlerRootView>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
