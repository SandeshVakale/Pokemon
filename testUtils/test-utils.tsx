import React, {PropsWithChildren} from 'react';
import {render} from '@testing-library/react';
import type {RenderOptions} from '@testing-library/react';
import {configureStore} from '@reduxjs/toolkit';
import type {PreloadedState} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import {RootState} from '../src/store';

import {listModel} from '../src/store/list';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: any;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({reducer: {listModel: listModel}, preloadedState}),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  function Wrapper({children}: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
