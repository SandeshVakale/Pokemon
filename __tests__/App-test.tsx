/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import {renderHook} from '@testing-library/react-hooks';
import * as Font from 'expo-font';
import {
  useFonts,
  Lexend_100Thin,
  Lexend_300Light,
  Lexend_400Regular,
  Lexend_500Medium,
  Lexend_600SemiBold,
  Lexend_700Bold,
  Lexend_800ExtraBold,
} from '@expo-google-fonts/lexend';
import {renderWithProviders} from '../testUtils/test-utils';
jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

jest.mock('expo-font', () => ({
  __esModule: true,
  ...jest.requireActual('expo-font'),
}));

it('renders App Lodaer correctly', async () => {
  await act(async () => {
    renderer.create(<App />).toJSON();
  });
});

it('renders App container correctly', async () => {
  // @ts-ignore:next-line
  jest.spyOn(Font, 'loadAsync').mockResolvedValue(true);

  const hook = renderHook(() =>
    useFonts([
      Lexend_100Thin,
      Lexend_300Light,
      Lexend_400Regular,
      Lexend_500Medium,
      Lexend_600SemiBold,
      Lexend_700Bold,
      Lexend_800ExtraBold,
    ]),
  );

  await hook.waitForNextUpdate();

  await act(async () => {
    const tree = await renderWithProviders(<App />);

    expect(tree).toMatchSnapshot();
  });
});
