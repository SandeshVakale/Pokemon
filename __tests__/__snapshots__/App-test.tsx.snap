// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App container correctly 1`] = `null`;
